src/main/java/org/elasticsearch/index/query/ExistsQueryBuilder.java

```diff
@@ -58,6 +85,87 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
         builder.endObject();
     }
 
+
+    @Override
+    public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
+        return newFilter(parseContext, name, queryName);
+    }
+
+    public static Query newFilter(QueryParseContext parseContext, String fieldPattern, String queryName) {
+        final FieldNamesFieldMapper fieldNamesMapper = (FieldNamesFieldMapper)parseContext.mapperService().fullName(FieldNamesFieldMapper.NAME);
+
+        MapperService.SmartNameObjectMapper smartNameObjectMapper = parseContext.smartObjectMapper(fieldPattern);
+        if (smartNameObjectMapper != null && smartNameObjectMapper.hasMapper()) {
```

Comment:
I couldn't find out yet how to test this branch. I'll try to find out what it actually does and find a way to insert SmartNameObjectMapper into parseContext in out setup.